staking/apr/compute.go

```diff
@@ -87,25 +94,25 @@ func ComputeForValidator(
 	)
 
 	if err != nil {
-		return &zero, err
+		return &zero, ErrInsufficientEpoch
 	}
 
 	blockNumAtOneEpochAgo := shard.Schedule.EpochLastBlock(oneEpochAgo.Uint64())
-
 	headerOneEpochAgo := bc.GetHeaderByNumber(blockNumAtOneEpochAgo)
+
 	if block.Header() == nil || headerOneEpochAgo == nil {
 		utils.Logger().Debug().
 			Msgf("apr compute headers epochs ago %+v %+v %+v",
 				oneEpochAgo,
 				blockNumAtOneEpochAgo,
 				headerOneEpochAgo,
 			)
-		return &zero, errors.New("can't get headers for APR computation")
+		return nil, ErrCouldNotRetreiveHeaderByNumber
```

Comment:
errors.Wrapf(ErrCouldNotRetreiveHeaderByNumber, "num header wanted %d", blockNumAtOneEpochAgo)

Reply:
in general, the error message should be adding info making it easy to fix/debug. Because this error will be propagated upward  as well. At each level propagation of error, make it easy to fix and see where things went wrong.